usaspending_api/etl/operations/treasury_appropriation_account/update_agencies.py

```diff
@@ -0,0 +1,69 @@
+from django.db import connections, DEFAULT_DB_ALIAS
+from usaspending_api.references.constants import DOD_CGAC, DOD_SUBSUMED_CGAC
+
+
+def update_treasury_appropriation_account_agencies():
+    """
+    Updates the awarding and funding toptier agencies in the TreasuryAppropriationAccount table.
+
+    For ATA:
+        - If awarding_toptier_agency_id is for a DOD subsumed agency (Army, Navy, Air Force), remap it to 097 (DOD).
+        - Otherwise attempt to look up the awarding agency by awarding_toptier_agency_id.  Nothing else we can
+          really do here.
+
+    For AID:
+        - If agency_id is for a DOD subsumed agency (Army, Navy, Air Force), remap it to 097 (DOD).
+        - If agency_id is a valid CGAC toptier agency, map it to toptier agency using the agency_id.
+        - If fr_entity_code is a valid FREC toptier agency, map it to toptier agency using the fr_entity_code.
+        - If agency_id is a shared toptier agency (011, 033, etc), use the frec_agency_cgac_association field in
+          the frec table to find the correct agency for the TAS.
+    """
+    subsumed = tuple(DOD_SUBSUMED_CGAC)
+    sql = f"""
```

Comment:
At cursory glance, it looks like the only difference between these sql statements is the name of the table being updated. This strikes me as an opportunity for code reuse.

Reply:
Already done.  I wouldn't bother reviewing this too much.  A few things changed on Friday (including a minor logic change).  Hopefully I'll check in a more finished version soon.